pkg/logbridge/logbridge.go

```diff
@@ -0,0 +1,116 @@
+package logbridge
+
+import (
+	"bufio"
+	"io"
+	"time"
+
+	"github.com/square/p2/pkg/logging"
+)
+
+// Copy implements a buffered copy operation between dest and src.
+// It returns the number of dropped messages as a result of insufficient
+// capacity
+// The caller is responsible for closing & flushing the reader and writer.
+func Copy(dest io.Writer, src io.Reader, capacity int, logger logging.Logger) {
+	lines := make(chan []byte, capacity)
+	droppedMessages := make(chan []byte, capacity)
+	go func(dest io.Writer, src io.Reader, lines chan []byte) {
```

Comment:
Considering the length of `Copy()`, the file as a whole would read better if you split these async functions into their own top-level functions.  This one right here would make a nice self-contained function instead of an anonymous lambda.  That would also let you document precisely what this routine is doing.